pkg/plugins/runtime/gateway/cluster_generator.go

```diff
@@ -199,15 +202,22 @@ func newClusterBuilder(
 // to a cluster can be different depending on the listener and the hostname,
 // we can't just build a cluster using the service name and tags, we have to
 // take the full configuration into account.
-func buildClusterResource(dest *route.Destination, c *clusters.ClusterBuilder) (*core_xds.Resource, error) {
+func buildClusterResource(
```

Comment:
from the comment

> take the full configuration into account

I think this is no longer true.

Reply:
done